### PR TITLE
Test add char string long char string

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem
`CHAR_STRING` and `LONG_CHAR_STRING` are not part of `TestCluster` and does not have specific tests

Writing the tests I noticed that when the spec says `max length 32` the current implementation count the `32` characters including the null termination character. I tend to think that in this case the spec meaning is `32 chars` + null terminated characters. So I have added some tests but disabled for this case until it is fixed. @bzbarsky-apple @tcarmelveilleux Am I understanding the spec correctly or does the null terminator included in the string length ?

#### Change overview
 * Add `CHAR_STRING` and `LONG_CHAR_STRING` to `Testcluster.yaml`
 * Add some tests into `src/app/tests/suites/TestCluster.yaml`
 
#### Testing
 * I have added some tests into `src/app/tests/suites/TestCluster.yaml`

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
> I tend to think that in this case the spec meaning is 32 chars + null terminated characters

Spec does not assume null-termination or any other length-storage strategy.  "32" in spec means bytes of string data.

File: src/app/tests/suites/TestCluster.yaml [hunk 1]
```diff
@@ -626,6 +626,72 @@ tests:
...
+      attribute: "char_string"
+      arguments:
+          value: "☉TestValueLongerThan10☉"
+
+    - label: "Read attribute CHAR_STRING"
+      command: "readAttribute"
+      attribute: "char_string"
+      disabled: true
+      response:
+          value: "☉TestVal"
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/TestCluster.yaml:661]: This is pretty surprising to me.  I would have expected a too-long write to fail entirely, not just write whatever fits...

If the spec does not define behavior clearly here, please file a spec issue.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/tests/suites/TestCluster.yaml (hunk 1 above)
Comment (vivien-apple) [reply src/app/tests/suites/TestCluster.yaml:661]: https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/3558

**Comment by vivien-apple** [comment]
Interestingly `Tests / Test Suites - Darwin` is failing because of: `Run command failure: OS Error 2055 (0x00000807): No buffer space available` . I will re-run the CI test and see if it happens again but there is definitively something wrong somewhere with network buffers.

**Review by jmartinez-silabs — APPROVED** [review]

**Comment by erjiaqing** [comment]
> #### Problem
> `CHAR_STRING` and `LONG_CHAR_STRING` are not part of `TestCluster` and does not have specific tests
> 
> Writing the tests I noticed that when the spec says `max length 32` the current implementation count the `32` characters including the null termination character. I tend to think that in this case the spec meaning is `32 chars` + null terminated characters. So I have added some tests but disabled for this case until it is fixed. @bzbarsky-apple @tcarmelveilleux Am I understanding the spec correctly or does the null terminator included in the string length ?
> 
> #### Change overview
> * Add `CHAR_STRING` and `LONG_CHAR_STRING` to `Testcluster.yaml`
> * Add some tests into `src/app/tests/suites/TestCluster.yaml`
> 
> #### Testing
> * I have added some tests into `src/app/tests/suites/TestCluster.yaml`

The spec says

> Note that the character string type is a bounded sequence of characters whose size bound format is not specified in the data model, but rather a property of the underlying encoding. Therefore, no assumptions are to be made about the presence or absence of a length prefix or null-terminator byte, or other implementation considerations.

Seems it is a intended behavior that not to specify how to terminate the string. The length of string is determined by the length only.

**Comment by bzbarsky-apple** [comment]
> Interestingly Tests / Test Suites - Darwin is failing because of: Run command failure: OS Error 2055 (0x00000807): No buffer space available

This is still happening, at least with Tsan....

**Comment by vivien-apple** [comment]
I have not been able to reproduce locally yet. The error from the logs is: 
```
[1625803633131] [18896:67722] CHIP: [IN] Sending msg 0x7b6400000558 to 0x0000000000BC5C01 at utc time: 1284475 msec[0m
[1625803633131] [18896:67722] CHIP: [IN] Sending secure msg on generic transport[0m
[1625803633131] [18896:67722] CHIP: [IN] Secure msg send status OS Error 2055 (0x00000807): No buffer space available[0m
[1625803633131] [18896:67722] CHIP: [-] OS Error 2055 (0x00000807): No buffer space available at ../../../examples/chip-tool/third_party/connectedhomeip/src/app/ReadClient.cpp:148[0m
[1625803633131] [18896:67722] CHIP: [DMG] Client[1] moving to [UNINIT][0m
[1625803633131] [18896:67722] CHIP: [TOO] TestCluster: OS Error 2055 (0x00000807): No buffer space available[0m
[1625803633131] [18896:67722] CHIP: [TOO] Run command failure: OS Error 2055 (0x00000807): No buffer space available[0m
```

I have rebased and force push, hoping that's something that has been fixed... Otherwise I need to find a way to reproduce locally.

**Comment by vivien-apple** [comment]
Hm. I'm seeing the same error with some other PRs that have landed on ToT. So we are probably close to some sort of limit and my PR is triggering it more often because of the large CHAR_STRING/LONG_CHAR_STRING messages.

**Pull request merged**